infrastructure/code/sast_enabled_test.go

```diff
@@ -39,119 +38,47 @@ func TestIsSastEnabled(t *testing.T) {
 		ApiError:    nil,
 	}
 
+	mockedSastResponse := snyk_api.SastResponse{
+		SastEnabled: true,
+		LocalCodeEngine: snyk_api.LocalCodeEngine{
+			AllowCloudUpload: false,
+			Url: "http://local.engine",
+			Enabled: true,
+		},
+	}
+
 	scanner := &Scanner{
 		SnykApiClient: apiClient,
 		errorReporter: error_reporting.NewTestErrorReporter(),
 		notifier:      notification.NewNotifier(),
 	}
 
 	t.Run("should return false if Snyk Code is disabled", func(t *testing.T) {
-		apiClient.ApiError = nil
-		apiClient.CodeEnabled = false
-		apiClient.LocalCodeEngineEnabled = false
-		config.CurrentConfig().SetSnykCodeEnabled(false)
+		mockedSastResponse.SastEnabled = false
 
-		enabled := scanner.isSastEnabled()
+		enabled := scanner.isSastEnabled(mockedSastResponse)
 
 		assert.False(t, enabled)
 	})
 
-	t.Run("should call the API to check enablement if Snyk Code is enabled", func(t *testing.T) {
-		apiClient.ApiError = nil
-		config.CurrentConfig().SetSnykCodeEnabled(true)
-
-		scanner.isSastEnabled()
-
-		assert.Equal(t, 1, len(apiClient.Calls))
-	})
-
-	t.Run("should return true if Snyk Code is enabled and the API returns true", func(t *testing.T) {
+	t.Run("should return true if Snyk Code is enabled", func(t *testing.T) {
 		config.CurrentConfig().SetSnykCodeEnabled(true)
-		apiClient.ApiError = nil
-		apiClient.CodeEnabled = true
-		apiClient.LocalCodeEngineEnabled = false
-
-		enabled := scanner.isSastEnabled()
+		mockedSastResponse.SastEnabled = true
+		enabled := scanner.isSastEnabled(mockedSastResponse)
 
 		assert.True(t, enabled)
 	})
 
-	t.Run("should return false if Snyk Code is enabled and the API returns false", func(t *testing.T) {
-		config.CurrentConfig().SetSnykCodeEnabled(true)
-		apiClient.ApiError = nil
-		apiClient.CodeEnabled = false
-		apiClient.LocalCodeEngineEnabled = false
-
-		enabled := scanner.isSastEnabled()
-
-		assert.False(t, enabled)
-	})
-
-	t.Run("should return false if Snyk Code is enabled and the API returns an error", func(t *testing.T) {
```

Comment:
Why is this test removed? Reaction to an API error should continue to be tested and be part of a refactoring.

Reply:
Added [here](https://github.com/snyk/snyk-ls/blob/e502a873af86b29fca7e1c4c61462d0e4c21c216/infrastructure/code/code_test.go#L754)